egui_glow/src/vao_emulate.rs

```diff
@@ -0,0 +1,59 @@
+#![allow(unsafe_code)]
+use glow::HasContext;
+
+pub(crate) struct BufferInfo {
+    pub location: u32, //
+    pub vector_size: i32,
+    pub data_type: u32, //GL_FLOAT,GL_UNSIGNED_BYTE
+    pub normalized: bool,
+    pub stride: i32,
+    pub offset: i32,
+}
+pub struct EmulatedVao {
+    buffer: glow::Buffer,
+    buffer_infos: Vec<BufferInfo>,
+    taught: bool,
+}
+impl EmulatedVao {
+    #[allow(clippy::needless_pass_by_value)]
+    pub(crate) fn new(buffer: glow::Buffer) -> Self {
+        Self {
+            buffer,
+            buffer_infos: vec![],
+            taught: false,
+        }
+    }
+    pub(crate) fn add_new_attribute(&mut self, buffer_info: BufferInfo) {
+        self.buffer_infos.push(buffer_info);
+    }
+    pub(crate) fn bind_vertex_array(&self, gl: &glow::Context) {
+        unsafe {
+            gl.bind_buffer(glow::ARRAY_BUFFER, Some(self.buffer));
+        }
+        for attribute in self.buffer_infos.iter() {
+            unsafe {
+                if !self.taught {
```

Comment:
`taught` is a strange word to use. it is also always `false`.

Reply:
i will remove it